src/physics/types.rs

```diff
@@ -1,14 +1,42 @@
 use geometry::types::{Point, Vector};
+use super::force::{Gravity, Attractor};
 use std::cmp::Eq;
-use std::collections::HashMap;
+
+// Environment ///////////////////////////////////////////////////////////////
+//
+// An environment represents a space in which bodies interact with fields.
+
+pub struct Environment {
+    pub bodies: Vec<Body>,
+    pub fields: Vec<Box<Field>>,
+}
+
+impl Environment {
+    pub fn new() -> Environment {
+        let field = BruteForceField::new();
+        Environment {
+            bodies: vec![],
+            fields: vec![Box::from(field)],
+        }
+    }
+
+    pub fn update(&mut self) {
+        for field in self.fields.iter() {
+            let forces = field.forces(&self.bodies[..]);
+
+            for (body, force) in self.bodies.iter_mut().zip(forces.iter()) {
+                body.apply_force(force);
+            }
+        }
+    }
+}
 
 // Body //////////////////////////////////////////////////////////////////////
 //
-// A body represents a moveable object in space.
+// A body represents a movable object in space.
 
 #[derive(Debug)]
 pub struct Body {
-    pub id: u32,
```

Comment:
why dont u need an id of a body anymore? how can we then identify a body ?

Reply:
I don't think we need to identify the body directly. Since the bodies are always stored in a `Vec` (in `Environment`), and that list is immutable, we can just use the index of the body to identify it.